api/api.go

```diff
@@ -649,7 +680,70 @@ func procDetachBackend(c libnetwork.NetworkController, vars map[string]string, b
 		return nil, errRsp
 	}
 
-	err := sv.Leave(vars[urlCnID])
+	sb, errRsp := findSandbox(c, vars[urlCnID], byID)
+	if !errRsp.isOK() {
+		return nil, errRsp
+	}
+
+	err := sv.Leave(sb)
+	if err != nil {
+		return nil, convertNetworkError(err)
+	}
+
+	return nil, &successResponse
+}
+
+/******************
+ Sandbox interface
+*******************/
+func procGetSandbox(c libnetwork.NetworkController, vars map[string]string, body []byte) (interface{}, *responseStatus) {
+	if epT, ok := vars[urlEpID]; ok {
+		sv, errRsp := findService(c, epT, byID)
+		if !errRsp.isOK() {
+			return nil, endpointToService(errRsp)
+		}
+		return buildContainerResource(sv.Info().Sandbox()), &successResponse
+	}
+
+	sbT, by := detectSandboxTarget(vars)
+	sb, errRsp := findSandbox(c, sbT, by)
+	if !errRsp.isOK() {
+		return nil, errRsp
+	}
+	return buildContainerResource(sb), &successResponse
+}
+
+func procGetSandboxes(c libnetwork.NetworkController, vars map[string]string, body []byte) (interface{}, *responseStatus) {
+	var list []*sandboxResource
+
+	shortID, queryByPid := vars[urlCnPID]
+	if queryByPid {
+		// Return all the prefix-matching networks
+		l := func(sb libnetwork.Sandbox) bool {
+			if strings.HasPrefix(sb.Info().ID(), shortID) {
+				list = append(list, buildContainerResource(sb))
+			}
+			return false
+		}
+		c.WalkSandboxes(l)
+	} else {
+		for _, sb := range c.Sandboxes() {
+			list = append(list, buildContainerResource(sb))
+		}
+	}
```

Comment:
Yes, only for partial-id query we need it.
In case we want all the sandboxes, instead, we use existing API which returns all of them. We have similar APIs for Network and Endpoint.